vaibhav/cart/shopping_cart_spec.rb

```diff
@@ -0,0 +1,139 @@
+#shopping cart test cases
+
+load 'shopping_cart.rb'
+load 'inventory.rb'
+
+describe 'add_Products'  do
+
+  before do
+    @inv_object = Inventory.new
+    @givenArray = [["apple",120,5]]
+  end
+
+  context "when user put product information" do
+        it "add products to inventory" do
+          
+            expect_any_instance_of(Kernel).to receive(:gets).and_return("apple")
+            expect_any_instance_of(Kernel).to receive(:gets).and_return(120)
+            expect_any_instance_of(Kernel).to receive(:gets).and_return(5)
+            result = @inv_object.Add_Products([])
+            expect(result).to eq(@givenArray)
+      end
+    end 
+  end
+
+  describe 'addTo_cart'  do
+
+    before do
+      @cart_object = Shopping_cart.new
+      @givenArray = [["orange",150,5]]
+    end
+
+    context "when user put right product name" do
+          it "add product to cart" do
+           
+            expect_any_instance_of(Kernel).to receive(:gets).and_return("orange")
+            expect_any_instance_of(Kernel).to receive(:gets).and_return(5)
+            result = @cart_object.AddTo_cart([],[])
+            expect(result).to eq(@givenArray)
+        end
+      end 
+
+      context "when user give worng product name" do
+        it "does not add product to cart" do
+         
+          expect_any_instance_of(Kernel).to receive(:gets).and_return("orannge")
+          result = @cart_object.AddTo_cart([],[])
+          expect(result).to eq('enter a valid name')
+      end
+    end 
+  end
+  
+  
+    describe 'displayAll'  do
+
+      before do
+        @inv_object = Inventory.new
+        @givenArray = [["apple",120,5],["orange",150,10]]
+      end
+      
+      context "in display all method" do
+            it "is to check display all products" do
+              
+             result = @inv_object.DisplayAll([])
+             expect(result).to eq(@givenArray)
+        end
+      end 
+    end
+
+
+    describe 'display_cart'  do
+
+      before do
+        @cart_object = Shopping_cart.new
+        @givenArray =  [["apple",120,5]]
+      end
+
+      context "in display cart method" do
+            it "is to check display cart products" do
+
```

Comment:
'context' should have conditional statements and 'it' should have the result which you are expecting